README.md

```diff
@@ -1,2 +1,198 @@
+
+```js
+> localClient.start()
+Promise { <pending> }
+> localServer.start()
+Promise { <pending> }
+> somebody connected on ws://localhost:8000/localClient/asdf
+connected to ws://localhost:8000/localClient/asdf
+
+> localClient.send({ type: 1, requestId: 1, data: { protocolData: [] } }, 'ws://localhost:8000/localClient/asdf')
+undefined
+> server sees BTP packet from ws://localhost:8000/localClient/asdf { type: 1, requestId: 1, data: [] }
+> localServer.stop()
+Promise { <pending> }
+> localClient.stop()
+Promise { <pending> }
+>
+```
+
+## Built-in LetsEncrypt registration
+
+If instead of `listen` you specify `tls`, the server will listen for secure WebSockets on port 443.
+This will not work on your laptop, or on a PaaS service like Heroku; you need a server (VPS) with
+its own IPv4 address, and make sure 'btp.example.com' points to your server and DNS has propagated.
+
+Then, SSH into your server:
+
+```sh
+ssh root@btp.example.com
+```
+
+Then run this node script:
+
+```js
+new BtpSpider({ tls: 'btp.example.com' }, (peerId) => {}, (obj, peerId) => {})
+```
+
+## Super-simple connector
+
+This connector forwards all incoming BTP packets from one peer to another,
+taking just a tiny profit in the process. It's not meant as something actually
+useful, just an example to show you how you can use the connection callback,
+the message callback, and the send method.
+
+See `examples/superSimpleConnector.js`
+
+```js
+const BtpPacket = require('btp-packet')
+const BtpSpider = require('./src/spider')
+let peers = []
+const spider = new BtpSpider({ listen: 8000 }, (peerId) => {
+  console.log('conn', peerId)
+  peers.push(peerId)
+}, (obj, peerId) => {
+  console.log('msg', obj, peerId)
+  if (peers.length !== 2) {
+    console.error('one-to-one connector needs exactly two peers')
+    return
+  }
+  let otherPeerId = peers[0]
+  if (peerId === peers[0]) {
+    otherPeerId = peers[1]
+  }
+  if (obj.type === BtpPacket.TYPE_PREPARE) {
+    obj.data.amount = '' + (parseInt(obj.data.amount) + 1) // pocket the profit! :)
+  }
+  spider.send(obj, otherPeerId)
+})
+spider.start()
+```
+
+You can test this super-simple connector by connecting two clients to it:
+```js
+const BtpPacket = require('btp-packet')
+const crypto = require('crypto')
+const BtpSpider = require('./src/spider')
+
+const client1 = new BtpSpider({
+  name: 'client1',
+  upstreams: [
+    {
+      url: 'ws://localhost:8000',
+      token: 'asdf'
+    }
+  ]
+}, (peerId) => {
+  console.log(`connected to ${peerId}`)
+}, (obj, peerId) => {
+  console.log(`client sees BTP packet from ${peerId}`, obj)
+})
+const client2 = new BtpSpider({
+  name: 'client2',
+  upstreams: [
+    {
+      url: 'ws://localhost:8000',
+      token: 'asdf'
+    }
+  ]
+}, (peerId) => {
+  console.log(`connected to ${peerId}`)
+}, (obj, peerId) => {
+  console.log(`client sees BTP packet from ${peerId}`, obj)
+})
+client1.start().then(() => {
+  return client2.start()
+}).then(() => {
```

Comment:
I really think using async/await would make this code cleaner to read

Reply:
ok, I'll try it out